### PR TITLE
Get more github ids

### PR DESCRIPTION
**Pull request by NatalieEO**
First attempt to the github id by using the token which I think is the best way, but if that is unsuccessful then use the avatarurl we have stored for them and grab the user id from there so we can grab more of the GitHub ids. I'm also going through the users by using the user_profile table and not the token table since there are a few more user_profiles than tokens.

There are more users we need to ban/delete. You can look at the list by checking the bottom of this document https://docs.google.com/document/d/1CNFzhYbmn-WgvNQ98vq-Sl4-uDq7dGjICgPBSGD7n-g/edit

I'm not sure what to do about the test failure about the poms?

### CODE REVIEW

**Comment by denis-yuen** [comment]
> I'm not sure what to do about the test failure about the poms?
 
Update from the release branch, I think you need at least https://github.com/dockstore/dockstore/commit/fe59ceef93db97c8a4aa64264004ab2e6911abbf

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]
Seems worth a try, would normally ask for test but this  won't be long for this world

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 1]
```diff
@@ -760,36 +761,50 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
+
+                if (avatarUrl != null) {
+                    Matcher m = GITHUB_ID_PATTERN.matcher(avatarUrl);
+                    if (m.matches()) {
+                        String id = m.group(1);
+                        u.getUserProfiles();
+                        userProfile.onlineProfileId = id;
+                        List<Token> userTokens = tokenDAO.findGithubByUserId(u.getId());
+                        if (!userTokens.isEmpty()) {
+                            userTokens.get(0).setOnlineProfileId(id);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java:799]: I guess there's never more than one token, but if there were, you would need to set it for all of them. I guess we do have this pattern of assuming there's at most one throughout the code. But you could just use `forEach`

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 2]
```diff
@@ -760,36 +761,50 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
-                LOG.info(ex.getMessage(), ex);
+            if (!u.isBanned()) {
+                User.Profile userProfile = u.getUserProfiles().get(TokenType.GITHUB_COM.toString());
+                String avatarUrl = userProfile.avatarURL;
+
+                if (avatarUrl != null) {
+                    Matcher m = GITHUB_ID_PATTERN.matcher(avatarUrl);
+                    if (m.matches()) {
+                        String id = m.group(1);
+                        u.getUserProfiles();
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java:795]: Do you need this? You're calling `u.getUserProfiles` on line 788.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 3]
```diff
@@ -760,36 +761,50 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
-                LOG.info("Updated " + u.getUsername());
-            } catch (Exception ex) {
-                usersCouldNotBeUpdated.add(u);
-                LOG.info(ex.getMessage(), ex);
+            if (!u.isBanned()) {
+                User.Profile userProfile = u.getUserProfiles().get(TokenType.GITHUB_COM.toString());
+                String avatarUrl = userProfile.avatarURL;
+
+                if (avatarUrl != null) {
+                    Matcher m = GITHUB_ID_PATTERN.matcher(avatarUrl);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java:792]: I'd feel slightly more comfortable with a test, but I experimented with it and it looks good.

**Review by coverbeck — APPROVED** [review]
A few nits, none of which are worth doing unless you're doing another revision anyway.

**Review by coverbeck — APPROVED** [review]

**Pull request merged**